utils.groovy

```diff
@@ -959,4 +959,54 @@ def build_and_push_image(params = [:]) {
     return manifest_digest
 }
 
+def get_arch_image_digest(from) {
+    def node_image_digest = shwrapCapture(""" skopeo inspect --raw docker://${from} """)
+    def digest_arch_map = shwrapCapture(""" echo '${node_image_digest}' | jq -r '.manifests[] | \
+                                      select(.platform.os == \"linux\") | \"\\(.platform.architecture)=\\(.digest)\"'
+                        """.stripIndent())
+    def arch_digests = [:]
+    digest_arch_map.split("\\n").each { line ->
+        def (arch, digest) = line.tokenize("=")
+        arch_digests[arch] = digest
+    }
+    return arch_digests
+}
+
+def brew_upload(arches, release, repo, manifest_digest, extensions_manifest_digest, version, pipecfg) {
+    def brew_principal = pipecfg.brew.principal
+    def brew_profile = pipecfg.brew.profile
```

Comment:
Minor/optional: cool with this, but also since they're just used once, it seems fine to directly use e.g. `--owner ${pipecfg.brew.principal}` in the invocation below. Pure personal style though. :)